torch/lib/c10d/TCPStore.cpp

```diff
@@ -57,11 +56,174 @@ TCPStoreDaemon::~TCPStoreDaemon() {
   closeStopSignal();
 }
 
-void TCPStoreDaemon::join() {
+void BackgroundThread::join() {
   daemonThread_.join();
 }
 
-void TCPStoreDaemon::queryFds(std::vector<struct pollfd>& fds) {
+#ifdef _WIN32
+void BackgroundThread::initStopSignal() {
+  ghStopEvent_ = CreateEvent(NULL, TRUE, FALSE, NULL);
+  if (ghStopEvent_ == NULL) {
+    throw std::runtime_error(
+        "Failed to create the control pipe to start the "
+        "BackgroundThread run");
+  }
+}
+
+void BackgroundThread::closeStopSignal() {
+  CloseHandle(ghStopEvent_);
+}
+
+void BackgroundThread::stop() {
+  SetEvent(ghStopEvent_);
+}
+#else
+void BackgroundThread::initStopSignal() {
+  if (pipe(controlPipeFd_.data()) == -1) {
+    throw std::runtime_error(
+        "Failed to create the control pipe to start the "
+        "BackgroundThread run");
+  }
+}
+
+void BackgroundThread::closeStopSignal() {
+  for (auto fd : controlPipeFd_) {
+    if (fd != -1) {
+      ::close(fd);
+    }
+  }
+}
+
+void BackgroundThread::stop() {
+  if (controlPipeFd_[1] != -1) {
+    // close the write end of the pipe
+    ::close(controlPipeFd_[1]);
+    controlPipeFd_[1] = -1;
+  }
+}
+#endif
+
+// TCPStoreListener class methods
+TCPStoreWorkerDaemon::TCPStoreWorkerDaemon(int listenSocket)
+    : BackgroundThread(listenSocket) {
+  daemonThread_ = std::thread(&TCPStoreWorkerDaemon::run, this);
+}
+
+void TCPStoreWorkerDaemon::addCallback(
+    std::string key,
+    std::function<void(c10::optional<std::string>, c10::optional<std::string>)>
```

Comment:
nit: we can create an alias of this callback function with `using`.

Reply:
Thanks!